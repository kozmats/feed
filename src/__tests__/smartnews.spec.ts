import { sampleFeed } from "./setup";

describe("smartnews", () => {
  it("should generate a valid feed", () => {
    const actual = sampleFeed.smartnews();
    expect(actual).toMatchSnapshot();
  });
});
