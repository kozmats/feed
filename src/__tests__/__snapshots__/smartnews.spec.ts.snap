// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smartnews should generate a valid feed 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?>
<rss version=\\"2.0\\" xmlns:content=\\"http://purl.org/rss/1.0/modules/content/\\" xmlns:snf=\\"http://www.smartnews.be/snf\\">
    <channel>
        <title>Feed Title</title>
        <link>http://example.com/</link>
        <description>This is my personnal feed!</description>
        <lastBuildDate>Sat, 13 Jul 2013 23:00:00 GMT</lastBuildDate>
        <docs>https://validator.w3.org/feed/docs/rss2.html</docs>
        <generator>https://github.com/jpmonette/feed</generator>
        <sf:logo>
            <url>http://exsample.com/logo.jpg</url>
        </sf:logo>
        <language>en</language>
        <image>
            <title>Feed Title</title>
            <url>http://example.com/image.png</url>
            <link>http://example.com/</link>
        </image>
        <copyright>All rights reserved 2013, John Doe</copyright>
        <category>Technology</category>
        <item>
            <title><![CDATA[Hello World]]></title>
            <link>https://example.com/hello-world</link>
            <snf:analytics><![CDATA[<script><!-- Analytics code --></script>]]></snf:analytics>
            <guid>https://example.com/hello-world</guid>
            <pubDate>Sat, 13 Jul 2013 23:00:00 GMT</pubDate>
            <description><![CDATA[This is an article about Hello World.]]></description>
            <content:encoded><![CDATA[Content of my item]]></content:encoded>
            <author>janedoe@example.com (Jane Doe)</author>
            <author>joesmith@example.com (Joe Smith)</author>
            <enclosure url=\\"https://example.com/hello-world.jpg\\" type=\\"image/jpeg\\"/>
        </item>
    </channel>
</rss>"
`;
